enhancements/ingress/operator-resource-configuration.md

```diff
@@ -0,0 +1,676 @@
+- Conversion webhooks handle any necessary translations
+
+### Version Skew Strategy
+
+#### Operator and API Server Skew
+
+During cluster upgrades, the API server may be updated before or after the ingress-operator:
+
+**Scenario 1**: API server updated first (has v1alpha1), operator not yet updated
+- v1alpha1 CRs accepted by API server
+- Old operator version ignores v1alpha1 fields (reads via v1 API)
+- No impact, custom resources wait for operator upgrade
+
+**Scenario 2**: Operator updated first (supports v1alpha1), API server not yet updated
+- Operator can handle v1alpha1 resources
+- API server doesn't serve v1alpha1 yet
+- Users continue using v1 API until API server updates
+
+**Maximum skew**: 1 minor version 
+
+### Operational Aspects of API Extensions
+
+#### Failure Modes
+
+1. **Invalid resource values**: 
+   - Rejected by Kubernetes validation
+   - User receives clear error message
+   - Operator continues with existing configuration
+
+2. **Controller failure**: 
+   - Operator deployment remains at current configuration
+   - Deployment status reflects error
+   - Operator logs provide debugging information
+
+3. **API conversion failure**: 
+   - Request fails with error message
+   - User notified of conversion issue
+   - Existing resources unaffected
+
+4. **Operator restart loop due to low resources**:
+   - Kubernetes backoff prevents rapid restarts
+   - Events and logs indicate resource pressure
+   - Admin can update IngressController to increase resources
+
+#### Support Procedures
+
+Standard OpenShift support procedures apply:
+
+**Gathering debug information**:
+```bash
+# View IngressController configuration
+oc get ingresscontroller default -n openshift-ingress-operator -o yaml
+
+# View operator deployment
+oc describe deployment ingress-operator -n openshift-ingress-operator
+
+# Check operator logs
+oc logs -n openshift-ingress-operator deployment/ingress-operator -c ingress-operator
+
+# Check pod resource usage
+oc adm top pod -n openshift-ingress-operator
+
+# Check QoS class
+oc get pod -n openshift-ingress-operator -o jsonpath='{.items[*].status.qosClass}'
+```
+
+**Common issues and resolutions**:
+- OOMKilled operator: Increase memory limits
+- CPU throttling: Increase CPU limits or reduce requests if not needed
+- Configuration not applied: Check operator logs for reconciliation errors
+
+## Implementation History
+
+- 2025-10-27: Enhancement proposed
+- TBD: Enhancement approved
+- TBD: API implementation merged to openshift/api
+- TBD: Controller implementation merged to cluster-ingress-operator
+- TBD: Feature available in Tech Preview (target: OpenShift 4.X)
+- TBD: Promotion to GA (target: OpenShift 4.Y, ~2 releases after Tech Preview)
+
+## Alternatives
+
+### Alternative 1: Configuration via ConfigMap
+
+Use a ConfigMap for operator resource configuration instead of API field.
+
+**Pros**:
+- Simpler to implement
+- No API version changes needed
+- Easy to update without CRD changes
```

Comment:
You would need a CRD change to add a reference to the ConfigMap... unless you would have the operator just check for a ConfigMap in `openshift-config` with some hard-coded name?